### ISSUE TITLE
Empty strings are output from method calls

### ISSUE THREAD

**Issue by joshuabaker** [nystudio107/craft-emptycoalesce#2]
I’ve encountered some unexpected behaviour when creating strings inline (i.e. not passing in via a variable).

An odd, example that I would’ve expected to work:

```twig
{% set foobar = '' ??? 'expected' %}
{{ dump(foobar) }}{# outputs string(0) "" #}
```

This becomes problematic when doing something like the following:

```twig
{% set foobar = craft.app.request.getQueryParam('empty_param') ??? 'expected' %}
{{ dump(foobar) }}{# outputs string(0) "" #}
```

It does work if I assign this to a variable first:

```twig
{% set empty = craft.app.request.getQueryParam('empty_param') %}
{% set foobar = empty ??? 'expected' %}
{{ dump(foobar) }}{# outputs string(8) "expected" #}
```

Is this something that can be adjusted to allow chaining without assignments?

**Comment by benface** [comment]
I have the exact same issue.

**Comment by khalwat** [comment]
I will have a look at the generated code, thanks

**Comment by khalwat** [comment]
Interesting, this compiles down to:

```php
        $context["foobar"] = (((true &&  !(null === ""))) ? ("") : ("expected"));
```

**Comment by khalwat** [comment]
Ahhhhh got it sorted!

**Comment by khalwat** [comment]
Fixed in https://github.com/nystudio107/craft-emptycoalesce/commit/b5d77c0f39449239a49a087b06b13567be762c29

Released in 1.0.2 -> https://github.com/nystudio107/craft-emptycoalesce/releases/tag/1.0.2

**Comment by joshuabaker** [comment]
This is impressive. Twig is complex af.

Thanks for sorting. 👍
